bundles/org.openhab.ui/web/src/components/cards/glance/location/status-badge.vue

```diff
@@ -222,11 +225,11 @@ export default {
       }
       return this.exprAst
     },
-    queryLightEquipment (equipmentType) {
-      let equipment = findEquipment(this.element.equipment, equipmentType, true)
+    queryLightEquipment () {
+      let equipment = findEquipment(this.element.equipment, 'Equipment_LightSource', true)
       if (!equipment.length) return []
       let allPoints = allEquipmentPoints(equipment)
-      let points = findPoints(allPoints, 'Point', true, 'Property_Light')
+      let points = findPoints(allPoints, 'Point_Control_Switch', false)
```

Comment:
In the previous code, as it does not look for a Switch, it will also find dimmer and color items. That is interpreted in line 203. Here it only looks for a switch.
I think we should look broader, but also avoid adding up multiple points from one equipment (e.g. having a Point_Control with a Dimmer item type and Point_Control_Switch with a Switch item type). In such case, the Switch item type should be used.

Reply:
> I think we should look broader, but also avoid adding up multiple points from one equipment (e.g. having a Point_Control with a Dimmer item type and Point_Control_Switch with a Switch item type). In such case, the Switch item type should be used.

I do basically agree with your statement :) .. however I don't know how to code it :( .. we would need to map each Item to the Thing that it comes from, and preferentially remove any duplications. => Perhaps you can kindly suggest some code?

Reply:
> I do basically agree with your statement :) .. however I don't know how to code it :( .. we would need to map each Item to the Thing that it comes from, and preferentially remove any duplications. => Perhaps you can kindly suggest some code?

I need to look into it in more detail. I also don't think we should make it dependent on the thing (you could have 2 items essentially representing the same light, but linked to a different thing for some obscure reason). It should only be looking at the semantic relationships. What about prefering Point_Control_Switch over Point_Control? That may just do.

When you are done working on this, I will have a look to improve it.